tests/sles4sap/hana_test.pm

```diff
@@ -0,0 +1,82 @@
+# SUSE's SLES4SAP openQA tests
+#
+# Copyright © 2018 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Checks HANA installation as performed by sles4sap/hana (sles4sap/hana_test)
+# Requires: sles4sap/hana, ENV variable SAPADM
+# Maintainer: Ricardo Branco <rbranco@suse.de>
+
+use base "sles4sap";
+use testapi;
+use strict;
+use utils 'ensure_serialdev_permissions';
+
+sub run {
+    my ($self) = @_;
+    my $ps_cmd = "ps auxw | grep HDB | grep -vw grep";
+
+    select_console 'root-console';
+
+    # The SAP Admin was set in sles4sap/hana
+    my $sapadmin = get_required_var('SAPADM');
+    my $sid = uc(substr($sapadmin, 0, 3));
+
+    # Allow SAP Admin user to inform status via $testapi::serialdev
+    assert_script_run "chown $sapadmin /dev/$testapi::serialdev";
+
+    type_string "su - $sapadmin\n";
+
+    # Change the working shell to bash as SAP's installer sets the admin
+    # user's shell to /bin/csh and csh has problems with strings that start
+    # with ~ which can be generated by testapi::hashed_string() leading to
+    # unexpected failures of script_output() or assert_script_run()
+    type_string "exec bash\n";
+
+    my $output = script_output "sapcontrol -nr 00 -function GetVersionInfo";
+    die "sapcontrol: GetVersionInfo API failed\n\n$output" unless ($output =~ /GetVersionInfo[\r\n]+OK/);
+
+    $output = script_output "sapcontrol -nr 00 -function GetInstanceProperties | grep ^SAP";
+    die "sapcontrol: GetInstanceProperties API failed\n\n$output" unless ($output =~ /SAPSYSTEM.+SAPSYSTEMNAME.+SAPLOCALHOST/s);
+
+    $output =~ /SAPSYSTEMNAME, Attribute, ([A-Z][A-Z0-9]{2})/m;
+    die "sapcontrol: SAP administrator [$sapadmin] does not match with System SID [$1]" if ($1 ne $sid);
+
+    my $password = get_required_var('PASSWORD');
+    $output = script_output "hdbsql -j -d NDB -u SYSTEM -n localhost:30015 -p $password 'SELECT * FROM DUMMY'";
+    die "hdbsql: failed to query the dummy table\n\n$output" unless ($output =~ /1 row selected/);
+
+    $output = script_output "sapcontrol -nr 00 -function Stop";
+    die "sapcontrol: Stop API failed\n\n$output" unless ($output =~ /Stop[\r\n]+OK/);
+
+    $output = script_output "sapcontrol -nr 00 -function StopService";
+    die "sapcontrol: StopService API failed\n\n$output" unless ($output =~ /StopService[\r\n]+OK/);
+
+    script_run "$ps_cmd | wc -l ; $ps_cmd";
```

Comment:
Would you mind turning that into an `assert_script_run`? We do not expect this call to fail, right? :)